sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1240,6 +985,236 @@ def _update_blocklist_on_error(
+                _add_to_blocked_resources(
+                    blocked_resources,
+                    launchable_resources.copy(region=region.name, zone=None))
+            elif code == 'type.googleapis.com/google.rpc.QuotaFailure':
+                # TPU VM pod specific error.
+                if 'in region' in message:
+                    # Example:
+                    # "Quota 'TPUV2sPreemptiblePodPerProjectPerRegionForTPUAPI'
+                    # exhausted. Limit 32 in region europe-west4"
+                    _add_to_blocked_resources(
+                        blocked_resources,
+                        launchable_resources.copy(region=region.name,
+                                                  zone=None))
+                elif 'in zone' in message:
+                    # Example:
+                    # "Quota 'TPUV2sPreemptiblePodPerProjectPerZoneForTPUAPI'
+                    # exhausted. Limit 32 in zone europe-west4-a"
+                    _add_to_blocked_resources(
+                        blocked_resources,
+                        launchable_resources.copy(zone=zone.name))
+
+            elif 'Requested disk size cannot be smaller than the image size' in message:
+                logger.info('Skipping all regions due to disk size issue.')
+                _add_to_blocked_resources(
+                    blocked_resources,
+                    launchable_resources.copy(region=None, zone=None))
+            elif 'Policy update access denied.' in message or code == 'IAM_PERMISSION_DENIED':
+                logger.info(
+                    'Skipping all regions due to service account not '
+                    'having the required permissions and the user '
+                    'account does not have enough permission to '
+                    'update it. Please contact your administrator and '
+                    'check out: https://skypilot.readthedocs.io/en/latest/cloud-setup/cloud-permissions/gcp.html\n'  # pylint: disable=line-too-long
+                    f'Details: {message}')
+                _add_to_blocked_resources(
+                    blocked_resources,
+                    launchable_resources.copy(region=None, zone=None))
+            elif 'is not found or access is unauthorized' in message:
+                # Parse HttpError for unauthorized regions. Example:
+                # googleapiclient.errors.HttpError: <HttpError 403 when requesting ... returned "Location us-east1-d is not found or access is unauthorized.". # pylint: disable=line-too-long
+                # Details: "Location us-east1-d is not found or access is
+                # unauthorized.">
+                _add_to_blocked_resources(
+                    blocked_resources,
+                    launchable_resources.copy(zone=zone.name))
+            else:
+                logger.warning(f'Got an unknown error: {message}; blocking '
+                               f'resources by its zone {zone.name}')
```

Comment:
From the screenshots in https://github.com/skypilot-org/skypilot/pull/2681#issuecomment-1870941879 this line duplicates already printed messages. Can skip?

Reply:
Good call. Removed. Thanks!